runtime/testy/studentAPI.py

```diff
@@ -137,19 +153,25 @@ def is_running(self, fn):
 
     return fn in self._coroutines_running
 
-  def _check_params(self, uid, param):
+  def _check_write_params(self, uid, param):
+    device_type = uid >> 72
+    valid_params = self.deviceName_to_writeParams[SENSOR_TYPE[device_type]]
+    if param not in valid_params:
+      raise StudentAPITypeError("Invalid param passed in, valid parameters for this device are: " + "".join(valid_params))
+
+  def _check_read_params(self, uid, param):
     device_type = uid >> 72
-    valid_params = self.deviceType_to_validParams[device_type]
+    valid_params = self.deviceName_to_readParams[SENSOR_TYPE[device_type]]
     if param not in valid_params:
       raise StudentAPITypeError("Invalid param passed in, valid parameters for this device are: " + "".join(valid_params))
 
   def _check_value(self, param, value):
     valid_values = self.param_to_valid_values[param]
-    if not isinstance(value, valid_values[0]):
+    if not type(value) in valid_values[0]:
```

Comment:
i vote for `isinstance`
http://stackoverflow.com/questions/1549801/differences-between-isinstance-and-type-in-python/1549854#1549854
The second argument can be a *tuple* of args 
http://stackoverflow.com/questions/33311258/python-check-if-isinstance-any-type-in-list
Note that this means we need `valid_values[0]` to be an argument, not a list.
since we shouldnt be overriding them anyway, i feel changing them to all be tuples is fine.